src/pattern-matching/infallible.md

```diff
@@ -0,0 +1,63 @@
+---
+minutes: 5
+---
+
+# Irrefutable Patterns
+
+In day 1 we briefly saw how patterns can be used to _destructure_ compound
+values. Let's review that and talk about a few other things patterns can
+express:
+
+```rust,editable
+fn takes_tuple(tuple: (char, i32, bool)) {
+    let a = tuple.0;
+    let b = tuple.1;
+    let c = tuple.2;
+
+    // This does the same thing as above.
+    let (a, b, c) = tuple;
+
+    // Ignore the first element, only bind the second and third.
+    let (_, b, c) = tuple;
+
+    // Ignore everything but the last element.
+    let (.., c) = tuple;
+}
+
+fn main() {
+    takes_tuple(('a', 777, true));
+}
+```
+
+<details>
+
+- All of the demonstrated patterns are _irrefutable_, meaning that they will
+  always match the value on the right hand side.
```

Comment:
I would say something here about how irrefutable patterns are still type-specific, e.g. `(_, _)` only matches pairs and `()` only matches the unit type.

Reply:
Good point, I added a note about that.